### PR TITLE
COCOS-49 - Pass agent configuration and computation via vsock

### LINKED ISSUE #49: Add agent configuration on vsock
https://github.com/ultravioletrs/cocos/blob/200f20767319138dc65e49a7a13387a73084ddf1/init/systemd/cocos-agent.service#L9C37-L11

Agent needs notifications URL in the environment. This will need to be defined when building the image (HAL).

### PR DESCRIPTION
**Pull request by SammyOina**
Resolves #49 
Requires #61 
- Added ability to read agent configurations directly from virtual socket connections.
- Streamlined computation configuration by removing outdated structures and adopting a new data model.
- Updated main agent application to utilize new configuration reading method.
- Adjusted manager service logic to align with the new computation and agent configuration model.

### CODE REVIEW

File: cmd/agent/main.go [hunk 1]
```diff
@@ -111,30 +97,29 @@ func newService(ctx context.Context, logger mglog.Logger, eventSvc events.Servic
 	return svc
 }
 
-// extractComputationValue to extract computation value from the command line.
-func extractComputationValue() (agent.Computation, error) {
-	cmdLineBytes, err := os.ReadFile("/proc/cmdline")
+func readConfig() (agent.Computation, error) {
+	l, err := vsock.Listen(manager.ManagerPort, nil)
```
Comment (drasko) [inline cmd/agent/main.go:101]: Why `l`? This is confusing. Why not `vs` if this is vsocket?

**Review by drasko — CHANGES_REQUESTED** [review]

File: cmd/agent/main.go (hunk 1 above)
Comment (SammyOina) [reply cmd/agent/main.go:101]: `l` for listener

Comment (drasko) [reply cmd/agent/main.go:101]: Yep, all clear - thanks.

**Review by drasko — APPROVED** [review]
LGTM

**Pull request merged**